runtime/print.c

```diff
@@ -55,8 +58,22 @@ static void printTagged(FILE *fp, TaggedPtr p, int style, struct PrintStack *sta
             fprintf(fp, "%" PRId64, taggedToInt(p));
             break;
         case TAG_FLOAT:
-            // XXX fix precision
-            fprintf(fp, "%g", taggedToFloat(p));
+            d = taggedToFloat(p);
+            if (isnan(d)) {
+                fprintf(fp, "NaN");
+                break;
+            }
+            if (isinf(d)) {
+                fprintf(fp, d < 0.0l ? "-Infinity" : "Infinity");
+                break;
+            }
+            if (d == 0.0) {
+                fprintf(fp, signbit(d) ? "-0.0" : "0.0");
+                break;
+            }
+            char buf[30]; // buf[25] should be enough, 30 just to be safe
```

Comment:
How do you know 25 is enough?

Reply:
I tested a few million random doubles with `emyg_dtoa`, max len was 25.

It was not specified, but they have [tested everything with 17 significant digits](https://github.com/miloyip/dtoa-benchmark#procedure). 

Wikipidea says:
> The 53-bit significand precision gives from 15 to 17 significant decimal digits precision. If a decimal string with at most 15 significant digits is converted to IEEE 754 double-precision representation, and then converted back to a decimal string with the same number of digits, the final result should match the original string.

1 sign + 17 digits + 1 `.` + 1 `e` + 1 exp sign + 3 exp = 24 so it makes sense.

Reply:
I am astounded that this is not documented in the source, which makes me question the dependability of this implementation.

Reply:
The benchmark testing code appears to use a buffer size of 1024.